app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.java

```diff
@@ -491,6 +498,84 @@ public void onLoadMore(int position) {
+            return;
+        }
+
+        NotificationViewData notification = notifications.getPairedItem(position);
+        if(!(notification instanceof NotificationViewData.Concrete)) {
+            if(notification == null) {
+                Log.e(TAG, String.format(
+                        "Tried to access notification but got null at position: %d of %d",
+                        position,
+                        notifications.size() - 1)
+                );
+            } else {
+                Log.e(TAG, String.format(
+                        "Expected NotificationViewData.Concrete, got %s instead at position: %d of %d",
+                        notification.getClass().getSimpleName(),
+                        position,
+                        notifications.size() - 1
+                ));
+            }
+
+            return;
+        }
+
+        StatusViewData.Concrete status = ((NotificationViewData.Concrete) notification).getStatusViewData();
+        if(status == null) {
+            Log.e(TAG, String.format(
+                    "Tried to access status in notification but got null at position: %d of %d",
+                    position,
+                    notifications.size() - 1)
+            );
+            return;
+        }
+
+        StatusViewData.Concrete updatedStatus = new StatusViewData.Builder(status)
+                .setCollapsed(isCollapsed)
+                .createStatusViewData();
+
+        NotificationViewData.Concrete concreteNotification = (NotificationViewData.Concrete) notification;
+        NotificationViewData updatedNotification = new NotificationViewData.Concrete(
+                concreteNotification.getType(),
+                concreteNotification.getId(),
+                concreteNotification.getAccount(),
+                updatedStatus,
+                concreteNotification.isExpanded()
+        );
+        notifications.setPairedItem(position, updatedNotification);
+        adapter.updateItemWithNotify(position, updatedNotification, false);
+
+        // Since we cannot notify to the RecyclerView right away because it may be scrolling
+        // we run this when the RecyclerView is done doing measurements and other calculations.
+        // To test this is not bs: try getting a notification while scrolling, without wrapping
+        // notifyItemChanged in a .post() call. App will crash.
+        recyclerView.post(() -> adapter.notifyItemChanged(position, notification));
+    }
+
+    /**
+     * Called when the status {@link ToggleButton} responsible for collapsing long
+     * status content is interacted with.
+     *
+     * @param isCollapsed Whether the status content is shown in a collapsed state or fully.
+     * @param position    The position of the status in the list.
+     */
+    @Override
+    public void onNotificationContentCollapsedChange(boolean isCollapsed, int position) {
```

Comment:
Not sure if it makes sense to copy javadoc for overrides

Reply:
oh yeah I just smash Alt+Ins, "Implement..." and then smash enter. It turned out useful to understant which method I was working on, I am in the process of removing it from the various places where this has happened rn